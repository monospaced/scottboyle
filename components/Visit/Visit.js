import React from "react";

const Visit = ({ project: { link, title } }) => {
  return (
    <p>
      <strong>Visit:</strong> <a href={link}>{title}</a>
    </p>
  );
};

export default Visit;
