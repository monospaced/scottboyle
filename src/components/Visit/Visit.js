import React from "react";

const Visit = ({ project: { link, title } }) => {
  return link && title ? (
    <p>
      <strong>Visit:</strong> <a href={link}>{title}</a>
    </p>
  ) : null;
};

export default Visit;
