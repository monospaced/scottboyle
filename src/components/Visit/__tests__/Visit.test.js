import React from "react";
import { render } from "enzyme";

import Visit from "../Visit.js";

describe("Visit component", () => {
  it("should render correctly", () => {
    const component = render(<Visit project={{}} />);
    expect(component).toMatchSnapshot();
  });

  it("should render a link", () => {
    const component = render(
      <Visit project={{ link: "https://site.com", title: "Site" }} />,
    );
    expect(component).toMatchSnapshot();
  });
});
