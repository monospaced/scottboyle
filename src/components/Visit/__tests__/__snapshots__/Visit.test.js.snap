// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visit component should render a link 1`] = `
<p>
  <strong>
    Visit:
  </strong>
   
  <a
    href="https://site.com"
  >
    Site
  </a>
</p>
`;

exports[`Visit component should render correctly 1`] = `null`;
